### ISSUE TITLE
Feature Request: Do not pollute arguments

### ISSUE THREAD

**Issue by osterman** [mumoshu/variant#81]
## what
* Do not pollute arguments with commands that conflate the purpose of the cli

```
Usage:
  helmctl [command]

Available Commands:
  build       Create a single executable from the Variantfile
  env         Print currently selected environment
  help        Help about any command
  init        Create a Variant command
  tiller      Commands that operate on the helm tiller
  version     Print the version number of this command
```

## why
* We've written a `helmctl` command to make it easier to initialize the helm tiller.
* The `env` and `build` commands don't apply or make sense to the cli tool (as determined by the author); I understand their value or purpose and don't know how to reconcile this desire to hide them.

## references
* https://github.com/cloudposse/geodesic/pull/465

**Comment by mumoshu** [comment]
Good point! I think `build` should be exposed only when you run `variant` directly against the variant command, like `VARIANTFILE=path/to/yourcommand variant build`.

Not sure about what we should do for `env`. Perhaps somehow making inclusions of builtin subcommand more explicit, like:

```yaml
builtins:
  env:
    enabled: true
```

**Comment by osterman** [comment]
Out of curiosity what happens if I define tasks that have the same names as the built ins? (And then mark those as private)

**Comment by mumoshu** [comment]
@osterman It redefines and hides the original implementation.

**Comment by osterman** [comment]
@mumoshu it appears to *not* hide the orginal in the `help` output. In the `help` output it just displays both.

The suggested `builtins` works for me. Alternatively, if I just redefine a built-in, I think it should override.

**Comment by mumoshu** [comment]
I just remembered that we already have `import: <url>` to import a specific task from any source.
So instead of adding a yet another config `builtins`, we can just enhance it to accept `import: variant://env` so that `tasks: {"env": {"import": "variant://env"}}` imports the builtin `env` task. And you can also rename it as you like.

WDYT?

**Comment by mumoshu** [comment]
> it appears to not hide the orginal in the help output. In the help output it just displays both.

This seems like a bug! Thanks for confirming. I'll take a look asap

**Comment by mumoshu** [comment]
> Alternatively, if I just redefine a built-in, I think it should override.

Opened #84 for this.

**Comment by mumoshu** [comment]
Let's enable #92 whenever the command name isn't `variant` i.e. you're using `variant` as the runtime of your command with shebang. Context: https://github.com/mumoshu/variant/pull/91#pullrequestreview-249344685
